Run pre-commit's black hook on the entire tree

superm1 opened the pull request:
In order for pre-commit to be useful, everything needs to meet a common baseline.

## Motivation

I want to be able to rely upon pre-commit to help fixup style when submitting a PR.  I also think it is worth having a pre-commit hook for the project for CI to mention style issues later.  I do this in another project (fwupd).

## Technical Details

I ran `pre-commit run --all-files` and this is the result for just the Python files.

## Test Plan

No testing is done as it's style only changes.

## Submission Checklist

- [x] Look over the contributing guidelines at https://github.com/ROCm/ROCm/blob/develop/CONTRIBUTING.md#pull-requests.

solaiys commented on projects/rocm-core/rdhc/rdhc.py:
projects/rocm-core/rdhc/rdhc.py :   LGTM.
No impact on the functionalities. Tested with this Black's style changes.

feizheng10 commented:
We had internal discussion about Black and other choices. Please ping me internally. Thanks!!

vedithal-amd commented:
rocprofiler-compute uses ruff for formatting checks and I believe pre-commit hook using block throughout the tree might conflict with ruff based formatting for rocprofiler-compute. cc @feizheng10

superm1 commented:
> rocprofiler-compute uses ruff for formatting checks and I believe pre-commit hook using block throughout the tree might conflict with ruff based formatting for rocprofiler-compute. cc @feizheng10 

In that case we should add exceptions to the hook. Can you list out the files that shouldn't be formatted with black?

jamessiddeley-amd commented:
[AMD Official Use Only - AMD Internal Distribution Only]

Hi Mario,

Rocprof-compute ruff formats the following files under these directories:

pythonpath = [
    ".",
    "src",
    "src/rocprof_compute_soc",
    "src/utils",
    "src/rocprof_compute_analyze/utils",
    "tests"
]

So these won’t need to be formatted by black since rocprof-compute is already formatted with ruff.

Let me know if you have any more questions!

Thanks,
James

From: Mario Limonciello ***@***.***>
Sent: Friday, November 28, 2025 10:08 AM
To: ROCm/rocm-systems ***@***.***>
Cc: Siddeley, James ***@***.***>; Review requested ***@***.***>
Subject: Re: [ROCm/rocm-systems] Run pre-commit's black hook on the entire tree (PR #1972)

Caution: This message originated from an External Source. Use proper caution when opening attachments, clicking links, or responding.

[Image removed by sender.]superm1 left a comment (ROCm/rocm-systems#1972)<https://github.com/ROCm/rocm-systems/pull/1972#issuecomment-3589657763>

rocprofiler-compute uses ruff for formatting checks and I believe pre-commit hook using block throughout the tree might conflict with ruff based formatting for rocprofiler-compute. cc @feizheng10<https://github.com/feizheng10>

In that case we should add exceptions to the hook. Can you list out the files that shouldn't be formatted with black?

—
Reply to this email directly, view it on GitHub<https://github.com/ROCm/rocm-systems/pull/1972#issuecomment-3589657763>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/BSGZICPT64RPSJ4N5N2SBJL37BQLZAVCNFSM6AAAAACM2DTY66VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZTKOBZGY2TONZWGM>.
You are receiving this because your review was requested.Message ID: ***@***.***>

superm1 committed:
Exclude rocprofiler-compute from pre-commit black hook

Signed-off-by: Mario Limonciello (AMD) <superm1@kernel.org>

superm1 committed:
Run pre-commit's black hook on the entire tree

In order for pre-commit to be useful, everything needs to meet a common
baseline.

Signed-off-by: Mario Limonciello (AMD) <superm1@kernel.org>

superm1 commented:
> So these won’t need to be formatted by black since rocprof-compute is already formatted with ruff.
> Let me know if you have any more questions!

Rebased the branch on `develop`, added an exclusion for that directory this time.

dayatsin-amd reviewed (approved):
Ok for changes in projects/rocr-runtime directory
